flask_rebar/request_utils.py

```diff
@@ -52,22 +52,19 @@ def __getitem__(self, key):
         return self.headers[key]
 
 
-def get_json_from_resp(resp):
-    return json.loads(resp.data.decode("utf-8"))
-
-
 def response(data, status_code=200, headers=None):
     """
     Constructs a flask.jsonify response.
 
     :param dict data: The JSON body of the response
     :param int status_code: HTTP status code to use in the response
+    :param dict headers: Additional headers to attach to the response
     :rtype: flask.Response
     """
-    resp = jsonify(data)
-
-    if not get_json_from_resp(resp=resp):
-        resp.data = ""
+    if data is None:
+        resp = Response(mimetype="application/json")
```

Comment:
This will enforcing a content-type on schema-less empty responses.

Reply:
A couple quick thoughts before I run out of brain-power for the day 🤣  (@joeb1415 @airstandley let's chat tomorrow morning to sync up - @barakalon would of course love to get your continued input here and/or we can get an email chain going.  @sytten I think you might have the closest thing to a known real-world use-case here too.  And anybody else that's following along of course! :) )

1. Agreed, this will enforce a content-type on schema-less empty responses
2. Presumably we could soften this a bit by making it just a default (i.e., if `headers` includes a `Content-Type` header, parrot that, else use "application/json")
2a.  Maybe instead of hard-coded "application/json" here we could/should future-proof a little by defining a `default_content_type` (or if we want to be consistent with Werkzeug and Flask terminology we could call it `default_mimetype`) which could be used here and elsewhere where we have similar situation (e.g, in the new `swagger_generator_v3` there's currently hard-coded "application/json" as well)
3. Assuming we decide that is sufficient to address the issue at-hand, it does cause a change in behavior..  Prior, empty schema-less responses returned with a Content-Type of "text/html".  So we need to decide if we consider that previous "text/html" behavior a "bug" in which case we're fixing it, or if this would be a breaking change unless we go through extra contortions to match that old behavior.

Reply:
Good summary Rick, thanks! 

1. Yes, this enforces a content-type on empty responses, but this is already the case from werkzeug, we're just changing it to be json instead of html. FWIW, [HTTP spec](https://tools.ietf.org/html/rfc2616#section-7.2.1) isn't strict on this, saying only that we should include the header if there is content, but not enforcing it and not specifying if it should/should not be present if there is no content. I could go either way on this one.

2. I like the idea to define a `default_content_type`, I guess this would be set when instantiating the registry? Perhaps, if we have the default, then we can set it even on No Content responses, but if it is not set, then we could omit that response header.

3. I think this is fixing a bug, certainly `text/html` was a non-sensical default content type to be including before.

Reply:
Having trouble finding a clean way to remove the header. Even if we remove it from our constructed response, flask seems to run it thru a [response wrapper](https://github.com/pallets/werkzeug/blob/0.15.2/src/werkzeug/test.py#L1021) again, which will add the Content-Type header back in again, before delivering it.

Reply:
@joeb1415 I think its customary to leave the header, and I agree with your point - `text/html` is just as arbitrary as `application/json`. So I don't think its a big deal to got with `application/json`.

With that, +1 for for (2) in @RookieRick 's comment above.

Reply:
For the record I don't disagree with the point that "application/json" is as valid/arbitrary as "text/html" for an empty response. (And the RFC standard is not at all helpful on the matter.)

However if apps were breaking from getting "text/html" instead of "application/json" as the content-type header, then I think we have to assume changing "text/html" to "application/json" may also be a breaking change.

I definitely like this default_mimetype approach. Most people will ignore it and get a sensible default, but if anyone needs the content-type for empty responses to remain "text/hml" they can have that.

Reply:
Rerunning and stepping through a couple of tests on the latest, this should now even get us back to where we were if they don't specify anything (i.e., if nothing is specified for default, you'll get text/html for empty string).  🎉 

Planning to go ahead and push this out later today as another patch release on the assumption that our breaking change was short-lived enough (and anybody impacted by it would almost certainly have been following this conversation :) ) with the commitment to move to pre-release and/or development releases going forward.  Given the horse is already out of the barn and we're unbreaking, that seems preferable vs doing a major release for a breaking change that has been reverted...  Not what I'd call "ideal" but "good enough" :)

If that gives anybody excessive heartburn, let me know - thanks all!!